src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs

```diff
@@ -378,6 +378,60 @@ public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)
             }
         }
 
+        /// <summary>
+        /// Validates that the actual collection contains same items as expected collection. If the test fails, this will display:
+        /// 1. Count if two collection count are different;
+        /// 2. Missed expected collection item when found
+        /// </summary>
+        /// <param name="expected">The collection that <paramref name="actual"/> should contain same items as</param>
+        /// <param name="actual"></param>
+        /// <param name="comparer">The comparer used to compare the items in two collections</param>
+        public static void CollectionEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual, IEqualityComparer<T> comparer)
+        {
+            var actualItemCountMapping = new Dictionary<T, ItemCount>(comparer);
+            var actualCount = 0;
+            foreach (var actualItem in actual)
+            {
+                if (actualItemCountMapping.TryGetValue(actualItem, out var countInfo))
+                {
+                    countInfo.Original++;
+                    countInfo.Remain++;
+                }
+                else
+                {
+                    countInfo = new ItemCount(1, 1);
+                }
+
+                actualItemCountMapping[actualItem] = countInfo;
```

Comment:
Link to: [design-guidelines/struct](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/struct)

```
❌ DO NOT define mutable value types.
```

perhaps it would be more maintainable if you just have it be a class in this case and then you won't need to reassign the struct here each time the value is modified.

Reply:
Hi @maryamariyan , Fixed, thanks.